src/tests/functional/plugin/shared/src/snippets/mha.cpp

```diff
@@ -55,16 +58,17 @@ void MHABase::SetUp() {
     setInferenceType(prc);
 }
 
- void MHABase::init_thresholds() {
+void MHABase::init_thresholds() {
     // Note: Libxsmm calculates Exp in a slightly different way, so the abs values might differ a bit. Ticket: 130699
 #ifdef SNIPPETS_LIBXSMM_TPP
     abs_threshold = 1e-6;
 #endif
-    if (inType == ov::element::bf16)
+    auto infer_precision = configuration.at(ov::hint::inference_precision.name()).as<ov::element::Type>();
```

Comment:
Should `model_input` be completely ignored here? What if we don't have precision enforced, but the model itself already has f16 or bf16 precision?

Reply:
The test class is currently designed in such a way that inference precision is always set: please take a look at `setInferenceType` calls in `SetUp` method. So if the model has f16 or bf16 precision, but we don't force any precision via additional config, inference precision will be set to inType==outType=="f16 or bf16"